### PR TITLE
implementation of new type representation and conversion for DDF

### PR DESCRIPTION
**Pull request by namma**
References for problem statement and design:
[1] Data Type Support Across Adatao Stacks, https://docs.google.com/document/d/1Iq8cWsoQ7uY7HUUN0Mzg8dX3asSOhIRmw0CWVxGYslw/edit#heading=h.ytm1c0ic3kje
[2] [Design] Type Support DDF / PE / BA, https://docs.google.com/spreadsheets/d/1Tr4YDAVx9mdRbtz_gLdeFs5rBZ0bF7bs5NM-mQtzhQY/edit#gid=0
Related:
[3] Complex type for JSON data, https://docs.google.com/document/d/1h2CvHzg_vifN8qSYTL_AS6VbI9ZD6iDOQvf8fjx_KDo/edit#heading=h.kp99bp86f19z

Main JIRA tickets: 
https://adatao.atlassian.net/browse/PE-572: to support new type representation and conversion
https://adatao.atlassian.net/browse/PE-514: to support non-primitive type in DDF/PA and clients.
Related tickets are linked to these 2 main tickets.

This is breaking change in DDF. However, all dependencies are already updated to support this change via the following PRs:
#124 new implementation of getRandomSample (sample2ddf) to make it work with complex data type in DDF
#119  Standardizing output of sql2txt with all dependencies addressed
#114 Move back to old sql2txt for backward compatability 
#113 fix writing ddf to text used in sql2txt and fetchRows, head, etc.
#110 Support fetchRows, head, etc. for DDF with complex type columns
#107 complete flattenDDF with bugs fixed for corner cases plus more comprehensive tests
#105 Flatten DDF with struct columns
#100 Support struct type and array type in DDF to support json data

### CODE REVIEW

**Comment by khangich** [comment]
ok to test

**Comment by ctn** [comment]
@trulite @shiti what do you think of this from the perspective of ddf-flink?

**Comment by Shiti** [comment]
@ctn I have requested access to the google docs mentioned in the PR. Otherwise if someone can post the relevant sections here, it will be easier for us to understand this PR in that context.

File: core/src/main/java/io/ddf/analytics/AStatisticsSupporter.java [hunk 1]
```diff
@@ -183,35 +178,6 @@ public double getVectorCovariance(String xColumnName, String yColumnName) throws
...
-        return bins;
-      }
-    }
-    return null;
-  }
-*/
-
-
   private double parseDouble(String s) {
     mLog.info(">>>> parseDouble: " + s);
```
Comment (ctn) [inline core/src/main/java/io/ddf/analytics/AStatisticsSupporter.java]: Do we need/want this `mLog` anymore?

File: core/src/main/java/io/ddf/content/Schema.java [hunk 2]
```diff
@@ -398,24 +398,32 @@ public void setData(Object[] data) {
 
 
   /**
-   * The R concept of a column "type", such as STRING, INT, LONG, etc.
+   * The R concept of a column "type", such as STRING, INT, BIGINT, etc.
    */
   public enum ColumnType {
```
Comment (mbbui) [inline core/src/main/java/io/ddf/content/Schema.java]: PS: each engine may use different type system to represent their data, while this enum only assume java type.
Solution: we should make this become Abstract Class. Then for each engine we will subclass with specific implementation of type conversion.

Comment (mbbui) [inline core/src/main/java/io/ddf/content/Schema.java]: @Shiti what do you think?

Comment (Shiti) [inline core/src/main/java/io/ddf/content/Schema.java]: I think that would be better approach since different engines support different types. It will keep the DDF API cleaner and easier for supporting other engines.
